ci/run.sh

```diff
@@ -8,22 +8,22 @@ source conan/bin/activate
 if [[ "$(uname -s)" == 'Linux' ]]; then
     if [ "$CC" == "clang" ]; then
         # clang + Ubuntu don't like to run with UBSAN, but ASAN works
-        export CMAKE_OPTIONS="$CMAKE_OPTIONS -DCMAKE_CXX_FLAGS=\"-fsanitize=address\" -DCMAKE_C_FLAGS=\"-fsanitize=address\" -DCMAKE_EXE_LINKER_FLAGS=\"-fsanitize=address\" -DCMAKE_MODULE_LINKER_FLAGS=\"-fsanitize=address\""
+        export CMAKE_OPTIONS="$CMAKE_OPTIONS"
```

Comment:
Hi Robin.  I am not sure it is a good idea to disable the sanitizers. In the practice, it has been helping us to identify many issues the last couple of years. 

Do you intend to make this change permanent or are you just testing something?

I will take a look to the CI errors taking place in 0.27-maintenance

Reply:
The sanitisers are causing great trouble in two places (both concerning RemoteIo).
1. calling `gethostbyname()` in http.cpp
I can avoid that by using curl to handle http.
2. in RemoteIo::mmap and RemoteIo::populateFakeData()
I don't think curl can help with that.

For now the quick way to get the CI to turn green on 0.27-maintenance is to disable ASAN.  I really appreciate you working with @neheb on the clang-tidy changes.  Thank You very much.  I really want to get back to working on the book.  The last 3 weeks have been totally consumed by:
1. python/bash_tests
2. darktable/XmpProperties::propertyList() 
3. building on ARM/Xcode 6.1, stack-protector, build switches
4. Useless community discussion about ISOBMFF, CR3, HEIF
5. charset=Unicode
6. I'm sure there was something else

I still have the following to do in the book:
1. Previews
To be researched
2. 4 file formats are not finished:
Olympus, RAF, GIF, CRW
3. I/O
including RemoteIo (which appears to have ASAN issues).
4. XMPsdk
How does Exiv2 interact with the Adobe code?
5. Writing files
Andreas uses a mysterious/undocumented "sophisticated algorithm".
6. Complete lots of unfinished parts of the book
security, documentation, etc..
7. A general review of everything in the book.
8. Undoubtably something I haven't realised or thought about!

It would be great to:
1. Have a test harness for the three programs in the book
tvisitor.cpp, dmpf.cpp and args.cpp
2. Test tvisitor on more RAW images.
https://raw.pixls.us

The scope of the book is vast.   I'll get there.  I always do.  I'm worried that when the book is complete, I'm going to start receiving hundreds of emails about bugs, typos, omissions, questions and of course claiming that it violates unspecified patents.  3 people are demanding that I change the font and redraw all the graphics in SVG.

Alison and I will have a 5 day "mini-break" in a cottage in Wales next week.  We haven't been more that 10k from home since March 1.  We both need a break.

Reply:
Oh I see, so it is more about ASAN detecting stuff in things that are not relevant for the Exiv2 core but useful for testing the remote part. 

In that case, would be the following an option?:
- Have one additional CI job testing the remote stuff with the basic configuration (gcc, Release, Ubuntu)  and ASAN disabled. 
- The rest of the CI jobs could keep ASAN enabled without running the remote stuff. 

On the other hand, with this approach we would not be running the remote stuff in all the configurations ... But I have the feeling that it is not so important to do it in every configuration.

Reply:
My priority is the book.  I want to get the CI to be green because @LeoHsiao1 is working on the python/bash_tests #1215 on 0.27-maintenance.  The CI will help him.  And it's useful for other folks who have been sending 0.27-maintenance patches (compilerFlags etc).

I believe ASAN is already working on 'master'.  If we do 0.27.4, I will restore ASAN on 0.27-maintenance and get it working perfectly.

Reply:
Understood. Let's do it like that. You had already enough pain with this 😉

Reply:
I have opened an issue about my ASAN/RemoteIo discoveries: #1307